decrypt.go

```diff
@@ -19,6 +21,14 @@ var ErrUnsupportedAlgorithm = errors.New("pkcs7: cannot decrypt data: only RSA,
 // ErrNotEncryptedContent is returned when attempting to Decrypt data that is not encrypted data
 var ErrNotEncryptedContent = errors.New("pkcs7: content data is a decryptable data type")
 
+// RFC 4055, 4.1
+// The current ASN.1 parser does not support non-integer defaults so the 'default:' tags here do nothing.
```

Comment:
If the `default` tags don't do anything here and we still need the defaults, maybe set them before unmarshaling the object?

Reply:
Yep, that's better. Copied the code from original PR without giving too much of a critical eye to this part. I've set the default for the `HashFunc` in b1fb0e54346f2f8b218ca002427fe0eff22564f9 now. The others are not used atm.